### ISSUE TITLE
Changes to Shield Block Value not updating Shield Slam damage

### ISSUE THREAD

**Issue by Hobojoeofthehobocode** [magey/tbc-warrior#45]
Glyph of Deflection's on use Shield Block Value is not adding to the damage of Shield Slam.

[WoWCombatLog-040621_163619.txt](https://github.com/magey/tbc-warrior/files/6267698/WoWCombatLog-040621_163619.txt)

**Comment by magey** [comment]
Relevant lines from the log above, abbreviated to easy reading:

```
4/6 16:36:38.435  SPELL_CAST_SUCCESS,"Shield Slam"
4/6 16:36:38.435  SPELL_DAMAGE,"Shield Slam",593
4/6 16:36:44.752  SPELL_CAST_SUCCESS,"Shield Slam"
4/6 16:36:44.752  SPELL_DAMAGE,"Shield Slam",576
4/6 16:36:51.268  SPELL_CAST_SUCCESS,"Shield Slam"
4/6 16:36:51.268  SPELL_DAMAGE,"Shield Slam",576
4/6 16:36:57.401  SPELL_AURA_APPLIED,"Glyph of Deflection",0x1,BUFF
4/6 16:36:57.401  SPELL_CAST_SUCCESS,"Glyph of Deflection"
4/6 16:36:57.868  SPELL_CAST_SUCCESS,"Shield Slam"
4/6 16:36:57.868  SPELL_DAMAGE,"Shield Slam",590
4/6 16:37:04.152  SPELL_CAST_SUCCESS,"Shield Slam"
4/6 16:37:04.152  SPELL_DAMAGE,"Shield Slam",587
4/6 16:37:10.451  SPELL_CAST_SUCCESS,"Shield Slam"
4/6 16:37:10.451  SPELL_DAMAGE,"Shield Slam",589
4/6 16:37:16.452  SPELL_CAST_SUCCESS,"Shield Slam"
4/6 16:37:16.452  SPELL_DAMAGE,"Shield Slam",586
4/6 16:37:17.418  SPELL_AURA_REMOVED,"Glyph of Deflection",0x1,BUFF
```

**Comment by Hobojoeofthehobocode** [comment]
![image](https://user-images.githubusercontent.com/82062293/113786485-d1c2db80-96fe-11eb-9a18-6fb535d5c9fc.png)
Glyph of Deflection does add Shield Block Value for the purposes of incoming damage, so it isn't a matter of the on use not adding Shield Block Value at all.

**Comment by Bushmaniac** [comment]
In my very short experience on the beta, it seems like the damage added to block value is always lagging one update behind your actual value, if you equip two items in a row only one seems to give the added block value, but then you can proc crusader and it's added to the equation.
Unequiping and equipping the shield again seems to be the easiest way to update it to get back on track.
I guess same thing might apply to glyph of deflection.

**Comment by Cleavis** [comment]
@Bushmaniac description does appear to match current behavior (v2.5.1.38225)

![image](https://user-images.githubusercontent.com/19498073/113878051-43466c80-9787-11eb-8adc-983635a958d1.png)
[WoWCombatLog-040721_094808.txt](https://github.com/magey/tbc-warrior/files/6272077/WoWCombatLog-040721_094808.txt)

also near the start of the combat log: can pop glyph of deflection out of combat, equip shield block value gear, and then shield slam to gain the benefit of glyph of deflection. 

seems to be an issue with last change to stats not being applied (crusader may work to update shield slam stats because +strength gives +SBV)

**Comment by Bushmaniac** [comment]
@Cleavis 

Aye, that was my idea of what was happening with crusader too.
I think the best way to test this in a controlled environment is to remove weapon with crusader on it and apply Blessing of Kings as the glyph is enabled and see if the added damage is everlasting.

**Comment by Cleavis** [comment]
did some more playing around with swapping gear and talents.

it seems **SBV for Shield Slam only updates with changes to Strength** (or equipping a shield, or relogging). can anecdotally confirm [hobojoe's post](https://github.com/magey/tbc-warrior/issues/45#issuecomment-814480861) that blocking incoming damage [appears to work correctly](https://imgur.com/Oh4ijnZ) with glyph pop, but outgoing shield slam (and charsheet) does not-

repro: 1) get naked with a shield (i used poinard + elementium reinforced bulwark). 2) shield slam several times. 3) equip styleen's impeding scarab. 4) shield slam several times. 5) equip any piece of +str gear (i used quick strike ring). 6) shield slam several times.

2 and 4 will match. 6 will differ.

2. https://classic.warcraftlogs.com/reports/7qpRVNAgBmTXzf1h#fight=last&source=7&type=damage-done - 31 hits, min 432 max 447 avg 439.8
4. https://classic.warcraftlogs.com/reports/xkFXhNKytDT3bmBJ#fight=last&source=1&type=damage-done - 26 hits, min 432 max 447 avg 440.5
6. https://classic.warcraftlogs.com/reports/GLCKNmHwvzkcVnF9#fight=last&type=damage-done&source=4 - 22 hits, 457 min 472 max avg 465.1

can also change 5,6 to 5) unequip styleen's, 6) shield slam several times for a 6 that will match 2 and 4 instead.

this could explain why crusader proc (+str) updates shield slam dmg and why equipping t3 gloves in my test above (has +str) updated shield slam dmg. 

along the way, also noticed putting points into shield mastery (+10% sbv) does not update sbv on the charsheet immediately, but putting points into vitality (+2% str) updates sbv (and sta, str) immediately. similar behavior to equipping/unequipping styleen's or equipping/unequipping/popping glyph of deflection  - new sbv not reflected on charsheet. not sure if related, and lower priority, but would also consider a bug.

**Comment by Cleavis** [comment]
fixed in 2.5.1.38521
![image](https://user-images.githubusercontent.com/19498073/116638851-2db30580-a935-11eb-8b05-08911656a2eb.png)
